### PR TITLE
fix: Revert -> "Revert "chore: Chore team metadata table + isOrganization migration 

### PR DESCRIPTION
**Pull request by hariombalhara**
This is reverting the revert for @sean-brydon's PR and also fixing the migration by removing the unique constraint [slug, isOrganization] which was unnecessary

Unique constraint [slug, parentId] is what we need which is already there.

There are some additional instances that have come up by merge from main and which have been also updated to use the new schema.

Requirements:

calcom can be an Organization slug -> slug='calcom', parentId=null
calcom can be a sub-team slug in org1 -> slug='calcom', parentId="org1"
calcom can be a sub-team slug in org2 -> slug='calcom', parentId="org2"
calcom can't be a non-sub-team slug -> slug='calcom'; parentId=null - Conflicts with 1
calcom can't be another organization's slug -> slug='calcom'; parentId=null - Conflicts with 1

### CODE REVIEW

**Comment by hariombalhara** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #13700** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/13700?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/13700?utm_source=stack-comment).

File: apps/web/playwright/fixtures/orgs.ts
Comment (hariombalhara) [inline apps/web/playwright/fixtures/orgs.ts:60]: It's okay to use include in tests.

File: packages/prisma/migrations/20240117062434_move_org_metadata_to_org_settings/migration.sql
Comment (hariombalhara) [inline packages/prisma/migrations/20240117062434_move_org_metadata_to_org_settings/migration.sql:37]: cc @sean-brydon Added COALESCE to default to false for the cases where `isOrganizationConfigured` and `isOrganizationVerified` were not set in JSON.

**Review by joeauyeung — DISMISSED** [review]
Early review and one of the test cases are failing

## Test cases
- calcom can be an Organization slug -> slug='calcom', parentId=null ✅ 
- calcom can be a sub-team slug in org1 -> slug='calcom', parentId="org1" ✅ 
- calcom can be a sub-team slug in org2 -> slug='calcom', parentId="org2" ❌ 
- calcom can't be a non-sub-team slug -> slug='calcom'; parentId=null - Conflicts with 1 ✅ 
- calcom can't be another organization's slug -> slug='calcom'; parentId=null - Conflicts with 1 ✅

**Review by joeauyeung — COMMENTED** [review]
Overall love the changes. Made a few non blocking comments.

**Comment by hariombalhara** [comment]
@joeauyeung What's failing exactly in that calcom slug test?

It is working for me. I have 2 orgs with one team each having same slug(calcom)

image

image
image
image

File: packages/trpc/server/routers/viewer/organizations/adminUpdate.handler.ts
Comment (hariombalhara) [reply packages/trpc/server/routers/viewer/organizations/adminUpdate.handler.ts:75]: This might not be what's intended. It's possible that organizationSettings is set but not all props are in it are set. So, the result could be different. The intention is to fallback for each and every property of organizationSettings individually

File: packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts
Comment (hariombalhara) [reply packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts:54]: This one would have to fixed separately as it's connected to many files.

**Comment by hariombalhara** [comment]
Another day, another conflict. Resolved it.

**Review by joeauyeung — APPROVED** [review]
@hariombalhara I don't know why that was failing before but it's working now. I've already went through all the files but it would be good to get second pair of eyes on this


## Test Case
- calcom can be a sub-team slug in org2 -> slug='calcom', parentId="org2" ✅

**Review by sean-brydon — COMMENTED** [review]
Tested also - everything seems to work as expected

@calcom/foundation Can we have an approval from one of you guys before we merge this also? Due to the migrations and high risk etc

**Review by emrysal — APPROVED** [review]
Tested & code reviews, looks good to me 🆒

**Pull request merged**
